porting/halium_7-1/Configuring.rst

```diff
@@ -118,6 +118,218 @@ At the end of the file, append this::
+
+Download kernel source to backport from
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Create a directory (outside your halium source tree) for the kernel source from which you will pull the newer drivers::
+
+    mkdir ~/kernel-backports
+
+Now clone the kernel source for the version and branch you need (v4.2 in the example below) into this directory::
+
+    cd ~/kernel-backports
+    git clone https://kernel.googlesource.com/pub/scm/linux/kernel/git/next/linux-next -b v4.2
+
+.. Note::
+
+    Although there are other kernel versions besides v4.2 available (as witnessed by available version tags on `the webpage <https://kernel.googlesource.com/pub/scm/linux/kernel/git/next/linux-next>`_), the backport script is specifically tailored to backporting from version 4.2 and thus effectively limits you to this option.
+
+Run script and fix errors
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Navigate to your backport scripts directory and issue the command below (using Python2 as shown)::
+
+    python2 ./gentree.py --copy-list ./copy-list --integrate --clean --git-revision v4.2 ~/kernel-backports/linux-next ~/halium/kernel/[VENDOR]/[MODEL_NAME]
+
+You will get error messages at the end and if they do not mention the backport Makefile and/or Kconfig, this means that the script has aborted before completion. You will then need to determine the cause and retry. 
+
+If the final error message concerns the Makefile and includes info about having generated a file named ``Makefile.rej``, this means you will find information in this file about changes that did not complete successfully, but which you can apply yourself. These need to be completed before proceeding with the build.
+
+Apply security patch
+^^^^^^^^^^^^^^^^^^^^
+
+An additional `generic security patch <https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/patch/?id=8a7b081660857a80c3efc463b3da790c4fa0c801>`_ needs to be applied. 
+
+Edit kernel defconfig
+^^^^^^^^^^^^^^^^^^^^^
+
+Your kernel config file (defconfig) needs to be modified in order for the backported driver and protocol code to be activated.
+
+Start by locating all lines beginning with ``CONFIG_BT_`` and move these to the end of the file. Collecting them there makes the subsequent steps somewhat easier by helping to keep track of the changes you make.
+
+Next, deactivate all that are activated, *i.e.* do not have a leading ``#``, by inserting this leading ``#``. At the same time, for each one, add a corresponding one beginning with ``CONFIG_BACKPORT_BT_``, *e.g.*::
+
+    CONFIG_BT=y
+
+becomes::
+
+    #CONFIG_BT=y
+
+and then insert the corresponding line for backports::
+
+    CONFIG_BACKPORT_BT=y
+
+Some more configuration settings are necessary, depending on your device. You will likely need the settings listed here, but additional ones can also be necessary::
```

Comment:
I'm confused now. Above you describe very clear, precise steps how I collect old configs, disable them, make new ones. Perfectly clear. Now we introduce vague language ... 'additional ones can also be necessary' How do I find them? If it is the procedure you explained above then I would simply remove this part because it does away with the clarity that had been established so far.

Reply:
Well, in my experience, that is the nature of the process. I would prefer for it to be perfectly clear, but given differences between devices, it is not. I have succeeded with two devices now, and I can't see a clearer way of putting it. :(

Reply:
I reformulated a bit here. Hope this helps.

Reply:
Redid this removing ambiguity.